gprofiler/application_separators.py

```diff
@@ -0,0 +1,232 @@
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_name, _, arg_val = arg.rpartition("=")
+            if arg_val is not None:
+                return arg_val
+
+    return None
+
+
+def append_python_module_to_proc_wd(process: Process, module_name: str) -> str:
+    return f'{process.cwd().replace("/", ".").strip(".")}.{module_name}'
+
+
+class GunicornApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "gunicorn"
+
+    def is_supported(self, process: Process) -> bool:
+        # Either gunicorn (for example: /usr/local/bin/gunicorn) or python that runs gunicorn
+        # (For example: /usr/local/bin/python /usr/local/bin/gunicorn)
+        if "gunicorn" in process.cmdline()[0]:
+            return True
+        if len(process.cmdline()) >= 2:
+            return "gunicorn" in process.cmdline()[1]
+        return False
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # As of gunicorn documentation the WSGI module name most probably will come from the cmdline and not from the
+        # config file / environment variables (they added the option to specify `wsgi_app` only in 20.1.0)
+        for arg in process.cmdline():
+            if ":" in arg:
+                if _IP_PORT_RE.match(arg):
+                    continue
+
+                return append_python_module_to_proc_wd(process, arg)
+
+        logger.warning(f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module")
+        return None
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return "uwsgi" in process.cmdline()[0]
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_cli_arg_value(process.cmdline(), "-w")
+        if wsgi is not None:
+            return append_python_module_to_proc_wd(process, wsgi)
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return append_python_module_to_proc_wd(process, wsgi)
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
```

Comment:
Not logging the process here. Also, as suggested on the gunicorn class, you should log the commandline arguments so we'll have *something* to debg.